oci/private/push.bzl

```diff
@@ -98,6 +98,19 @@ oci_push(
 ```
 """
 
+# Helper rule for ensuring that the crane and yq toolchains are actually
+# resolved for the architecture we are targeting.
+def _transition_to_target_impl(settings, attr):
+    return {
+        "//command_line_option:extra_execution_platforms": settings["//command_line_option:platforms"],
```

Comment:
This didn't work for me on bazel 6.4.0, throwing an error like:

```
FATAL: bazel crashed due to an internal error. Printing stack trace:
java.lang.RuntimeException: Unrecoverable error while evaluating node 'com.google.devtools.build.lib.skyframe.RegisteredExecutionPlatformsValue$Key@98ab941b' (requested by nodes 'ToolchainContextKey{configurationKey=BuildConfigurationKey[6e2757b80890975f9114c79f8e56d8ad17b03e4d5656996857adff85802c3252], toolchainTypes=[ToolchainTypeRequirement{toolchainType=@rules_oci//oci:crane_toolchain_type, mandatory=true}], execConstraintLabels=[], forceExecutionPlatform=Optional.empty, debugTarget=false}')
	at com.google.devtools.build.skyframe.AbstractParallelEvaluator$Evaluate.run(AbstractParallelEvaluator.java:633)
	at com.google.devtools.build.lib.concurrent.AbstractQueueVisitor$WrappedRunnable.run(AbstractQueueVisitor.java:365)
	at java.base/java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(Unknown Source)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(Unknown Source)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(Unknown Source)
	at java.base/java.util.concurrent.ForkJoinPool.scan(Unknown Source)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(Unknown Source)
Caused by: java.lang.ClassCastException: class com.google.devtools.build.lib.cmdline.Label cannot be cast to class java.lang.String (com.google.devtools.build.lib.cmdline.Label is in unnamed module of loader 'app'; java.lang.String is in module java.base of loader 'bootstrap')
	at com.google.devtools.build.lib.skyframe.TargetPatternUtil.parseAllSigned(TargetPatternUtil.java:83)
	at com.google.devtools.build.lib.skyframe.RegisteredExecutionPlatformsFunction.compute(RegisteredExecutionPlatformsFunction.java:92)
	at com.google.devtools.build.skyframe.AbstractParallelEvaluator$Evaluate.run(AbstractParallelEvaluator.java:562)
	... 7 more
```

I had to force it to a string explicitly:
```suggestion
        "//command_line_option:extra_execution_platforms": str(settings["//command_line_option:platforms"][0]),
```

Reply:
Done, though I did add a loop to iterate over the platforms.